src/lib/homestore_backend/hs_homeobject.cpp

```diff
@@ -115,8 +122,25 @@ void HSHomeObject::init_homestore() {
     LOGI("Initialize and start HomeStore with app_mem_size = {}", homestore::in_bytes(app_mem_size));
 
     std::vector< homestore::dev_info > device_info;
-    for (auto const& path : app->devices()) {
-        device_info.emplace_back(std::filesystem::canonical(path).string(), homestore::HSDevType::Data);
+    bool has_data_dev = false;
+    bool has_fast_dev = false;
+    for (auto const& dev : app->devices()) {
+        auto input_dev_type = dev.type;
+        auto detected_type = get_device_type(dev.path.string());
+        LOGD("Device {} detected as {}", dev.path.string(), detected_type);
+        auto final_type = (dev.type == DevType::AUTO_DETECT) ? detected_type : input_dev_type;
+        if (final_type == DevType::UNSUPPORTED) {
+            LOGW("Device {} is not supported, skipping", dev.path.string());
+            continue;
+        }
+        if (input_dev_type != DevType::AUTO_DETECT && detected_type != final_type) {
+            LOGW("Device {} detected as {}, but input type is {}, using input type", dev.path.string(), detected_type,
+                 input_dev_type);
+        }
+        auto hs_type = (final_type == DevType::HDD) ? homestore::HSDevType::Data : homestore::HSDevType::Fast;
+        if (hs_type == homestore::HSDevType::Data) { has_data_dev = true; }
+        if (hs_type == homestore::HSDevType::Fast) { has_fast_dev = true; }
+        device_info.emplace_back(std::filesystem::canonical(dev.path).string(), hs_type);
     }
 
```

Comment:
NIT:  it `s better to add a sanity check here to see whether device_info is empty,  and log an critical error and exit if yes.  this will happen when all the devices are DevType::UNSUPPORTED